src/main/java/org/assertj/core/condition/VerboseCondition.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2021 the original author or authors.
+ */
+package org.assertj.core.condition;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import org.assertj.core.annotations.Beta;
+import org.assertj.core.api.Condition;
+
+/**
+ * Condition that shows the expected and the value to test in the verbose
+ * description. The description output of the value to test could also be
+ * transformed by functions.
+ *
+ * <pre><code class=
+ * 'java'> Condition&lt;String&gt; verboseCondition = VerboseCondition.verboseCondition(
+        actual -&gt; actual.length() &lt; 4, // predicate
+        "word must be shorter than &gt;4&lt;", // verbal description of the predicate
+        (s) -&gt; String.format("%s (original word: %s)", s.length(), s)); // actual value transformation
+ *</code></pre>
+ * 
+ * Tested against the (not matching) word `foooo` the description of the
+ * condition would be `word must be shorter than &lt;4 (maximum word length)&gt;
+ * but was &lt;5 (original word: foooo)&gt;`
+ * <p>
+ * 
+ * @param <T>        the type of object the given condition accept.
+ *
+ * @author Stefan Bischof
+ */
+@Beta
+public final class VerboseCondition<T> extends Condition<T> {
+
+  /**
+   * Creates a new <code>{@link VerboseCondition}</code>
+   *
+   * <pre><code class=
+   * 'java'> Condition&lt;String&gt; verboseCondition = VerboseCondition.verboseCondition(
+          actual -&gt; actual.length() &lt; 4, // predicate
+          "word must be shorter than &gt;4&lt;", // verbal description of the predicate
+          (s) -&gt; String.format("%s (original word: %s)", s.length(), s)); // actual value transformation
+   *</code></pre>
+   *
+   * @param <T>                  the type of object the given condition accept.
+   * @param predicate            the Predicate that tests the value to test.
+   * @param description          describes the Condition verbal.
+   * @param objectUnderTestDescription Function that is called to transforms the value
+   *                             to test into a text, when the actual value does
+   *                             not match the predicate.
+   * @return the created {@code VerboseCondition}.
+   * @throws NullPointerException if the predicate is {@code null}.
+   */
+  public static <T> VerboseCondition<T> verboseCondition(Predicate<T> predicate, String description,
+      Function<T, String> objectUnderTestDescription) {
+
+    return new VerboseCondition<>(predicate, description, objectUnderTestDescription);
+  }
+
+  private Function<T, String> objectUnderTestDescription;
+
+  private String description;
```

Comment:
is it possible to remove it? it does not seem to be used

Reply:
Line 103